MapboxMobileEventsTests/MMEAPIClientTests_cedar.mm

```diff
@@ -0,0 +1,327 @@
+#import <Cedar/Cedar.h>
+#import <Foundation/Foundation.h>
+#import "MMEAPIClient.h"
+#import "MMEConstants.h"
+#import "MMEEvent.h"
+#import "MMECommonEventData.h"
+#import "MMETrustKitWrapper.h"
+#import "MMENSURLSessionWrapperFake.h"
+#import "MMEAPIClientFake.h"
+
+using namespace Cedar::Matchers;
+using namespace Cedar::Doubles;
+
+@interface MMEAPIClient (Tests)
+
+@property (nonatomic) id<MMENSURLSessionWrapper> sessionWrapper;
+@property (nonatomic, copy) NSURL *baseURL;
+@property (nonatomic) BOOL usesTestServer;
+@property (nonatomic) NSBundle *applicationBundle;
+@property (nonatomic, copy) NSString *userAgent;
+
+- (void)setupUserAgent;
+
+@end
+
+SPEC_BEGIN(MMEAPIClientSpec)
+
+describe(@"MMEAPIClient", ^{
+    
+    __block MMEAPIClient *apiClient;
+    
+    beforeEach(^{
+        apiClient = [[MMEAPIClient alloc] initWithAccessToken:@"access-token"
+                                                userAgentBase:@"user-agent-base"
+                                               hostSDKVersion:@"host-sdk-1"];
+    });
+    
+    describe(@"- testing API setup", ^{
+        context(@"when MMETelemetryTestServerURL is nil", ^{
+            
+            beforeEach(^{
+                spy_on([NSUserDefaults standardUserDefaults]);
+                [NSUserDefaults standardUserDefaults] stub_method(@selector(objectForKey:)).with(MMETelemetryTestServerURL).and_return(nil);
+                
+                apiClient = [[MMEAPIClient alloc] initWithAccessToken:@"access-token"
+                                                        userAgentBase:@"user-agent-base"
+                                                       hostSDKVersion:@"host-sdk-1"];
+                
+                apiClient.sessionWrapper should_not be_nil;
```

Comment:
Consider moving this into an it block and maybe testing that it is the correct instance type